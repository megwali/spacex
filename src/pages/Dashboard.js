import styled from 'styled-components';
import LaunchCard from '../components/LaunchCard';
import useLaunchList from '../hooks/useLaunchList';

const CardList = styled.div`
  border: 1px solid lightgrey;
  margin: auto;
  max-width: 600px;
`;

const Dashboard = () => {
  const launches = useLaunchList();

  return !launches?.length ? 'loading...' : (
    <CardList>
      {launches.map(launch => (
        <LaunchCard launch={launch} />
      ))}
    </CardList>
  );
};

export default Dashboard;
