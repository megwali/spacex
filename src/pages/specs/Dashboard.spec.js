import { create } from 'react-test-renderer';
import Dashboard from '../Dashboard';

describe('Dashboard', () => {
  it ('renders', () => {
    const component = create(<Dashboard />);
    expect(component.toJSON()).toMatchSnapshot();
  });
});
