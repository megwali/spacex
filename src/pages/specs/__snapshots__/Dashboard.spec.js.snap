// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Dashboard renders 1`] = `
<div>
  Dashboard
</div>
`;
